### ISSUE TITLE
Replacing Slot Image at my vm

### ISSUE THREAD

**Issue by Sashasonic** [Kwoth/NadekoBot#2343]
Hi, this has never worked and have tried asking many a time in the discord Nadeko server, to no effect.  I've run the Nadeko self-hosted bot for quite a long while now, but this issue has never been fixed. I've also tried countless amount of times to 'search the Nadeko Support server' the latest fix I found I printed below.  As it was not directed at me - still not sure how to work with that.

Today with suggestions from the Nadeko server - **I uploaded a new pic without the spaces as suggested, then the latest command used was _sudo mv LLAPnadekoSlots.png ~/NadekoBot/src/NadekoBot/bin/Release/netcoreapp2.0/data/images/slots/background.png_** Also ran the .reloadimages cmd.

TIA,
Sasha

Would really appreciate a fix, as it's been a long while, and never use the slot because of it.

this person seemed to know how to fix: 

> 斜視-眼 - 04/12/2018
I know how
You need selfhost though
【NCBP】Potato - 04/12/2018
NadekoBot/src/NadekoBot/_strings/ResponseStrings.en-US.json
斜視-眼 - 04/12/2018
Go to the Images file
And then to slots
skymanplay - 04/12/2018
no
斜視-眼 - 04/12/2018
Then edit the image
Do NOT replace the image
skymanplay - 04/12/2018
default png slot image
斜視-眼 - 04/12/2018
Unless you want to fuck up your images.json
skymanplay - 04/12/2018
have someone?
斜視-眼 - 04/12/2018
Im telling you how
There is background.png
And background2.png
【NCBP】Potato - 04/12/2018
NadekoBot/src/NadekoBot/_strings/ResponseStrings.en-US.json @斜視-眼
skymanplay - 04/12/2018
i know
斜視-眼 - 04/12/2018

That is how you get this
Ty Potato
【NCBP】Hedgehog :Think: - 04/12/2018
@skymanplay you want the 

> default?

>

**Comment by Sashasonic** [comment]
Not 100% sure why, but yes I do need to use sudo with a lot of things.. it's only me that interacts with it tho.  In the example where it shows my 'last command'.. at least it did seem to 'move the file', but did not show up in server using the slot command.  

Which is why I think it may also need changing at background2.png (that's what I noticed in the above 'solution' by someone) Really just need to replace all 'copies' of the original nadeko one you'd think.. but don't know how.

**Comment by TwinDragon** [comment]
![discord_2018-06-05_15-06-52](https://user-images.githubusercontent.com/4114117/40999983-2b5deb02-68d2-11e8-9b8a-c627c4cd6be1.png)
Worked for me. Also, the command is **.imagesreload** now. I had to look that up real quick.

**Comment by Macley-Kun** [comment]
![image](https://user-images.githubusercontent.com/26381427/41001458-3df0aa22-6911-11e8-93a4-9f602df3fed3.png)
Works for me too. There are btw two locations where i post the images.
/NadekoBot/src/NadekoBot/data/images/slots
/NadekoBot/src/NadekoBot/bin/Release/netcoreapp2.0/data/images/slots

If you however, want to LOCK a file from being changed. You can use this command.
`chattr +i /root/.pm2/dump.pm2`
To unlock it, you do this:
`chattr -i /root/.pm2/dump.pm2`

I don't recommand this, because somewhere something goes wrong. But for the sake of fixing the issue for you, this would be a good workaround.

**Comment by Sashasonic** [comment]
So happy it works for everyone else.. lol I've had a few people 'try'..  and I have used 'imagesreload' as well still no effect. 

Will get to your information soon thanks @Macley-Kun  I'm also not getting an email for any updates here - so will fix that as well.

**Comment by Macley-Kun** [comment]
Btw you do stop the bot before you replace the images right?
Because that’s how I always do it.

The line I made of chattr is for Linux, but you can just navigate to the file you want to lock, do ‘pwd’ And copy that path with the full file name behind it in that command. It then locks the correct file. 

**but don’t use that unless everything else doesn’t seem to work**

**Comment by Sashasonic** [comment]
I don't want to lock anythng, I don't think.. Just need to get my Nadeko Slot image in the right place, but no, I've never 'stopped' by bot first as the guidance so far has been from the Nadeko support server and that was never mentioned. (my vm is ubuntu/linux)

**Comment by Macley-Kun** [comment]
Stop bot
Put your photos with the exact same name on these places 
/NadekoBot/src/NadekoBot/data/images/slots
/NadekoBot/src/NadekoBot/bin/Release/netcoreapp2.0/data/images/slots
Then start bot. That’s all to it.

**Comment by Sashasonic** [comment]
Would you mean to 'replace' the images with the same name (I don't use winscp, as I know that would be easier - but that's never connected / worked either) - so that makes working with my vm a lot trickier.. so, I have to use the (google cloud) ssh 'shell'.. popout thing (looks like a command window) so I remember to get the last image 'moved' it had to be 

sudo mv (something)

**Comment by Macley-Kun** [comment]
yeh i mean replace.
you indeed use mv to move something. like `mv /root/myimage.png /root/newfolder/`
With Winscp, i would try to get it to work, you don't have to but i would recommand getting that to work. Kinda strange if a mayor tech company wouldn't let users connect with ftp to their paid virtualmachines...

I'm not really sure, but is it possible to like.. Open a photo image with notepad locally, copy the content, and past that in a new file with the extention corresponding to the original photo image?

**Comment by Sashasonic** [comment]
I'll have a fiddle in a minute, I do appreciate the help.. the winscp just kept mucking up and not 'working' with the keys, I swear I did everything correctly - I have a computer 'tech' guy that I will get to work it out, if following your idea doesn't help me 

I am intending to get the guy's help in connecting via the winscp, but have been putting it off.

Thanks again
I'll let you know if/when fixed

**Comment by Macley-Kun** [comment]
np,

Btw if you would use putty to connect to your vm, winscp would import those settings. You could try (but then you should not be having a ssh session) use port 22. But again i'm not sure if that would work... Because normally i would expect google to keep things simple.

**Comment by Sashasonic** [comment]
putty or no, the metakeys didn't work, kept stuffing up.

**Comment by Macley-Kun** [comment]
Don't you mean the private key didn't work..?

To login you normally put (when creating a vm) a public key from puttygen (for example).
And use the private key you saved from puttygen to login automaticlly. But that's how i always do it. Maybe metakeys is some other methode..

**Comment by Sashasonic** [comment]
nvm

**Comment by TwinDragon** [comment]
When I did the replacement on mine, I didn't stop the bot. Just reloaded the images and it still worked. But that's besides the point.

I don't have FTP working on my server (something I have to fix still, just lazy), so I had to upload it to something else to download from. This is what I did step-by-step:

1. I upload the image temporarily somewhere.
2. I downloaded it to my Debian tower via SSH using PuTTY. (I use a username/password login, I haven't really bothered setting up my SSH key to it because its pretty locked down anyway)
3. mv background.png ./Bots/Discord/NadekoBot/src/NadekoBot/data/images/slots/background_n.png (I did the same with background2.png because at the time of testing, I didn't really check which was being used, but only background2.png is used anyway)
4. cd Bots/Discord/NadekoBot/src/NadekoBot/data/images/slots/
5. mv background.png background_old.png && mv background2.png background2_old.png (I kept the old ones as backup so I didn't delete them)
6. mv background_n.png background.png (same for the other)
7. Go into Discord (I used a private channel for this), issue **.imagesreload**, tested with a **.slot 5** and it worked.

My method probably was a little bit longer than necessary, but you get the idea. Also, note that none of my commands issued ever used **sudo**. It is best to not to use sudo unless absolutely necessary for things on any Linux install as no process should be running as root except for system-level stuff. Something like this should always remain at the user level:
![putty_2018-06-08_05-11-17](https://user-images.githubusercontent.com/4114117/41152859-8eecc4c2-6ada-11e8-93ba-3eadee11ba28.png)

Now, I don't know how yours is set up. I'm just going on the assumption basing on previous commands used which I can neither confirm nor deny if its just a simple permission issue that the bot can't read the file because of the fact its under a different user/group entirely.
